core/src/test/java/org/apache/calcite/sql/test/SqlEqualsDeepTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.test;
+
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.util.Litmus;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Unit test for fix [CALCITE-4402] The equalsDeep returns wrong result.
+ */
+class SqlEqualsDeepTest {
+
+  @Test void testCountEqualsDeep() throws SqlParseException {
+
+    SqlNode countExpression = parseExpression("count(a)");
+    SqlNode countDistinctExpression = parseExpression("count(distinct a)");
+
+    assertThat("count(a) doesn't equals count(a)",
+        countExpression.equalsDeep(parseExpression("count(a)"), Litmus.IGNORE));
```

Comment:
Please factor this to a method with two expression arguments.

The method should be like `assertThat("{" + expr1 + "}.equalsDeep({" + expr2 + "})", parseExpression(expr1).equals(parseExpression(expr2)), is(..))`